lib/guides_style_18f/navigation.rb

```diff
@@ -1,14 +1,67 @@
 # @author Mike Bland (michael.bland@gsa.gov)
 
+require 'jekyll'
 require 'safe_yaml'
 
 module GuidesStyle18F
   module FrontMatter
     def self.load(basedir)
-      Dir[File.join basedir, 'pages', '**', '*.md'].map do |f|
-        [f[basedir.size + 1..-1], SafeYAML.load_file(f, safe: true)]
+      init_file_to_front_matter_map(basedir).merge(
+        site_file_to_front_matter(init_site(basedir)))
+    end
+
+    def self.init_site(basedir)
+      Dir.chdir(basedir) do
+        config = SafeYAML.load_file('_config.yml', safe: true)
+        adjust_config_paths(basedir, config)
+        site = Jekyll::Site.new(Jekyll.configuration(config))
+        site.reset
+        site.read
+        site
+      end
+    end
+    private_class_method :init_site
+
+    def self.adjust_config_paths(basedir, config)
+      source = config['source']
+      config['source'] = source.nil? ? basedir : File.join(basedir, source)
+      destination = config['destination']
+      destination = '_site' if destination.nil?
+      config['destination'] = File.join(basedir, destination)
+    end
+    private_class_method :adjust_config_paths
+
+    def self.site_file_to_front_matter(site)
+      site_pages(site).map do |page|
+        [page.relative_path,  page.data]
       end.to_h
     end
+    private_class_method :site_file_to_front_matter
+
+    def self.site_pages(site)
+      pages = site.collections['pages']
+      if pages.nil?
+        site.pages.select do |page|
+          page.relative_path.start_with?('/pages') || page.url == '/'
+        end
+      else
+        pages.docs.each { |page| page.data['permalink'] ||= page.url }
```

Comment:
This block is only returning a collection of URLs for `page` objects, correct?  The above `site.pages.select` block is returning `page` objects though, right (based on the path/URL criteria defined inside the block)?  Or am I missing/misinterpreting something?